src/survey/factory.ts

```diff
@@ -0,0 +1,81 @@
+import { Factory } from 'src/common'
+import { Survey } from './survey'
+import { Labels, LanguageCode } from 'src/language'
+
+export type SurveyFactoryParams = {
+  ownerUuid: string
+  name: string
+  label?: string | null
+  languages?: LanguageCode[]
+  published?: boolean
+  draft?: boolean
+  collectUri?: string
+  descriptions?: Labels[]
+}
+export class SurveyFactory implements Factory<Survey> {
```

Comment:
check if you can't use a class to avoid creating a new class.
e.g. 
```
export const SurveyFactory:Factory<Survey>{...}
```

then you use it like:
```
SurveyFactory.createInstance(...)
```

Reply:
In this case, it will give error `Binding element 'createInstance' implicitly has an 'any' type.ts(7031)`